src/languages/cpp.js

```diff
@@ -7,6 +7,16 @@ Website: https://isocpp.org
 */
 
 function(hljs) {
+  function optional(s) {
+    return '(' + s + ')?';
+  }
+  var DECLTYPE_AUTO_RE = 'decltype\\(auto\\)'
+  var NAMESPACE_RE = '[a-zA-Z_]\\w*::'
```

Comment:
`hljs.UNDERSCORE_IDENT_RE + '::'` ?
Btw, I'm not 100% sure but looks like C++ allows underscore as the first character in identifiers, so maybe all (most?) instances of `hljs.IDENT_RE` below should really be `hljs.UNDERSCORE_IDENT_RE`. (This is not showstopper of course.)

Reply:
Pulling this thought out into it's own issue: https://github.com/highlightjs/highlight.js/issues/2353